skpro/distributions/log_normal.py

```diff
@@ -0,0 +1,178 @@
+        "capabilities:exact": ["mean", "var", "energy", "pdf", "log_pdf", "cdf", "ppf"],
+        "distr:measuretype": "continuous",
+    }
+
+    def __init__(self, mu, sigma, index=None, columns=None):
+
+        self.mu = mu
+        self.sigma = sigma
+        self.index = index
+        self.columns = columns
+
+        # todo: untangle index handling
+        # and broadcast of parameters.
+        # move this functionality to the base class
+        self._mu, self._sigma = self._get_bc_params()
+        shape = self._mu.shape
+
+        if index is None:
+            index = pd.RangeIndex(shape[0])
+
+        if columns is None:
+            columns = pd.RangeIndex(shape[1])
+
+        super(LogNormal, self).__init__(index=index, columns=columns)
+
+    def _get_bc_params(self):
+        """Fully broadcast parameters of self, given param shapes and index, columns."""
+        to_broadcast = [self.mu, self.sigma]
+        if hasattr(self, "index") and self.index is not None:
+            to_broadcast += [self.index.to_numpy().reshape(-1, 1)]
+        if hasattr(self, "columns") and self.columns is not None:
+            to_broadcast += [self.columns.to_numpy()]
+        bc = np.broadcast_arrays(*to_broadcast)
+        return bc[0], bc[1]
+
+    def energy(self, x=None):
+        r"""Energy of self, w.r.t. self or a constant frame x.
+
+        Let :math:`X, Y` be i.i.d. random variables with the distribution of `self`.
+
+        If `x` is `None`, returns :math:`\mathbb{E}[|X-Y|]` (per row), "self-energy".
+        If `x` is passed, returns :math:`\mathbb{E}[|X-x|]` (per row), "energy wrt x".
+
+        Parameters
+        ----------
+        x : None or pd.DataFrame, optional, default=None
+            if pd.DataFrame, must have same rows and columns as `self`
+
+        Returns
+        -------
+        pd.DataFrame with same rows as `self`, single column `"energy"`
+        each row contains one float, self-energy/energy as described above.
+        """
+        approx_spl_size = self.get_tag("approx_energy_spl")
+        approx_method = (
+            "by approximating the energy expectation by the arithmetic mean of "
+            f"{approx_spl_size} samples"
+        )
+        
+        # splx, sply = i.i.d. samples of X - Y of size N = approx_spl_size
+        N = approx_spl_size
+        if x is None:
+            splx = self.sample(N)
+            sply = self.sample(N)
+            # approx E[abs(X-Y)] via mean of samples of abs(X-Y) obtained from splx, sply
+            spl = splx - sply
+            energy = spl.apply(np.linalg.norm, axis=1, ord=1).groupby(level=1).mean()
+            energy = pd.DataFrame(energy, index=self.index, columns=["energy"])
+        else:
+            d = self.loc[x.index, x.columns]
+            mu_arr, sd_arr = d._mu, d._sigma
+            
+            c_arr = x*(2*self.cdf(x)-1)-2*exp((mu_arr+sd_arr**2)/2)*(self.cdf((np.log(x)-mu_arr-sd_arr**2)/sd_arr)+self.cdf(sd_arr/mu_arr**0.5)-1)
```

Comment:
could you kindly write down the formula in math, or explain otherwise how you are getting this expression?

Reply:
Hey, so this is basically the CRPS score which is E[IX-xI] -0.5E[|X-X'|] from which I was unable to isolate the first term. Wolfram alpha too wasnt able to produce a closed form for the integral. Can we change the description of the energy method accordingly or would you rather we go by the approximation in the base class?

Reply:
Why don't we try to spend a short discussion on trying to see whether we can get somewhere. Removing it entirely is always an option.

A "trick" to isolate the first term is to observe that adding the same constant to `x` and `X` (so, the location parameter) should leave the formula unchanged.

Regarding the integral, which integral concretely are you feeding into Wolfram Alpha?

Reply:
yes so this is what I got. If I try to add limits it crashes.
![Screenshot 2024-03-23 130300](https://github.com/sktime/skpro/assets/77113227/4bacce19-7229-47a3-bf3e-f5a4d88981ae)

Reply:
this looks correct. Now you need to add the limits. That should be an easy substitution, no? Use that

$\lim_{x\rightarrow -\infty} erf(x) = -1$, and $\lim_{x\rightarrow \infty} erf(x) = 1$. You need to be careful with the sign, but that should be it?

Reply:
moved discussion here: https://github.com/sktime/skpro/issues/219